### PR TITLE
Develop 4 --> Main

### PR DESCRIPTION
**Pull request by zainm97**
## Description

Merge develop branch for sprint 4 to main

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by LaithM96 — APPROVED** [review]
looks good, checks passed as well, good work!

File: terraform/environment/dev/resources/lambda/filterData/lambdaHandler/dataFilterLambda.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
+import { GetObjectCommand, PutObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { Readable } from 'stream';
+import csv from 'csv-parser';
+
+const GALLERI_ONS_BUCKET_NAME = process.env.BUCKET_NAME
+const GRIDALL_CHUNK_1 = process.env.GRIDALL_CHUNK_1
+const GRIDALL_CHUNK_2 = process.env.GRIDALL_CHUNK_2
+const GRIDALL_CHUNK_3 = process.env.GRIDALL_CHUNK_3
+
+const participatingIcbs = new Set([
```
Comment (msandhuNHS) [inline terraform/environment/dev/resources/lambda/filterData/lambdaHandler/dataFilterLambda.js:10]: Are these mocks or actual values needed in system?

Comment (zainm97) [reply terraform/environment/dev/resources/lambda/filterData/lambdaHandler/dataFilterLambda.js:10]: these are the set of participating icb, actually needed

Comment (msandhuNHS) [reply terraform/environment/dev/resources/lambda/filterData/lambdaHandler/dataFilterLambda.js:10]: Do you want throw them in a JSON in a data folder?

**Pull request merged**